hoot-core/src/main/cpp/hoot/core/conflate/tile/TileUtils.cpp

```diff
@@ -184,78 +146,42 @@ void TileUtils::writeTilesToOsm(
 
   int randomTileIndex = -1;
   if (selectSingleRandomTile)
-  {
     randomTileIndex = TileUtils::getRandomTileIndex(tiles, randomSeed);
-  }
 
+  OsmMapPtr boundaryMap = tilesToOsmMap(tiles, nodeCounts, randomTileIndex, selectSingleRandomTile);
+
+  OsmMapWriterFactory::write(boundaryMap, outputPath);
+  OsmMapWriterFactory::writeDebugMap(boundaryMap, "osm-tiles");
+}
+
+OsmMapPtr TileUtils::tilesToOsmMap(
+  const std::vector<std::vector<geos::geom::Envelope>>& tiles,
+  const std::vector<std::vector<long>>& nodeCounts,
+  int randomTileIndex, const bool selectSingleRandomTile)
+{
   OsmMapPtr boundaryMap(new OsmMap());
   int bboxCtr = 1;
   for (size_t tx = 0; tx < tiles.size(); tx++)
   {
     for (size_t ty = 0; ty < tiles[tx].size(); ty++)
     {
-      // TODO: This code could possibly be replaced by GeometryUtils::createMapFromBounds.
-      const geos::geom::Envelope env = tiles[tx][ty];
-      const double circularError = ConfigOptions().getCircularErrorDefaultValue();
-
-      NodePtr lowerLeft(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMinX(), env.getMinY()),
-          circularError));
-      boundaryMap->addNode(lowerLeft);
-      NodePtr upperRight(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMaxX(), env.getMaxY()),
-          circularError));
-      boundaryMap->addNode(upperRight);
-      NodePtr upperLeft(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMinX(), env.getMaxY()),
-          circularError));
-      boundaryMap->addNode(upperLeft);
-      NodePtr lowerRight(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMaxX(), env.getMinY()),
-          circularError));
-      boundaryMap->addNode(lowerRight);
-
-      WayPtr bbox(new Way(Status::Unknown1, boundaryMap->createNextWayId(), circularError));
-      bbox->addNode(lowerLeft->getId());
-      bbox->addNode(upperLeft->getId());
-      bbox->addNode(upperRight->getId());
-      bbox->addNode(lowerRight->getId());
-      bbox->addNode(lowerLeft->getId());
```

Comment:
Took the advice of the comment above and made a little change in the `GeometryUtils::createMapFromBounds` to remove this duplicated code and use it below.